hooks/useAlternateObservationZones.ts

```diff
@@ -0,0 +1,104 @@
+import {QueryClient, UseQueryResult, useQuery} from '@tanstack/react-query';
+import axios, {AxiosError} from 'axios';
+import {Logger} from 'browser-bunyan';
+import {safeFetch} from 'hooks/fetch';
+import {LoggerContext, LoggerProps} from 'loggerContext';
+import {useContext} from 'react';
+import {AlternateObservationZones, KMLData, KMLPlacemark} from 'types/nationalAvalancheCenter';
+import {xml2json} from 'xml-js';
+
+export const useAlternateObservationZones = (url: string): UseQueryResult<AlternateObservationZones, AxiosError> => {
+  const {logger} = useContext<LoggerProps>(LoggerContext);
+  const key = queryKey(url);
+  const thisLogger = logger.child({query: key});
+
+  return useQuery<AlternateObservationZones, AxiosError>({
+    queryKey: key,
+    queryFn: (): Promise<AlternateObservationZones> => fetchAlternateObservationZones(thisLogger, url),
+    enabled: !!url,
+    cacheTime: 60, // TODO: Change cache to one day after testing
+    staleTime: 60,
+    initialData: [],
+  });
+};
+
+export const prefetchAlternateObservationZones = async (queryClient: QueryClient, url: string, logger: Logger) => {
+  const key = queryKey(url);
+  const thisLogger = logger.child({query: key});
+  thisLogger.debug('prefetching alternate observation zones');
+
+  await queryClient.prefetchQuery({
+    queryKey: key,
+    queryFn: async (): Promise<AlternateObservationZones> => {
+      thisLogger.trace(`prefetching`);
+      const result = fetchAlternateObservationZones(thisLogger, url);
+      return result;
+    },
+    cacheTime: 60, // TODO: Change cache to one day after testing
+    staleTime: 60,
+  });
+};
+
+export const fetchAlternateObservationZones = async (logger: LoggerProps['logger'], url: string): Promise<AlternateObservationZones> => {
+  const thisLogger = logger.child({url: url});
+  thisLogger.debug('Fetching alternate observation zones');
+  const response = await safeFetch(() => axios.get<string>(url), thisLogger, 'alternate observation zones');
+  try {
+    const alternateZones = parseKmlData(response);
+    thisLogger.debug('Fetched alternate observation zones');
+    return alternateZones;
+  } catch (error) {
+    thisLogger.error({error: error}, 'Error parsing KML data');
+    throw new Error('Error parsing KML data');
+  }
+};
+
+function queryKey(url: string) {
+  return ['alternateZoneKML-', url];
+}
+
+export function parseKmlData(response: string): AlternateObservationZones {
+  const kmlDataJson = xml2json(response, {compact: true, spaces: 2});
+  const kmlData = JSON.parse(kmlDataJson) as KMLData;
```

Comment:
What we're doing in the rest of the codebase is using Zod to make sure that `JSON.parse()` spits out an object with the fields we're actually expecting. I think you already have a schema for `KMLData`, so we can `zod.safeParse()` the input and catch errors here instead of failing at runtime when we incorrectly access a field that we expected to exist but is missing.

Reply:
This is a really cool capability, thanks for pointing me towards it; I added it here and in my transform function. One approach at getting the feature collections to merge involved populating the feature properties with default values to make them compatible with the mapLayer. For this the safeParse was super helpful!